File: app/src/main/java/com/rmuhamed/sample/poketest/data/dao/PokemonDAO.java
package com.rmuhamed.sample.poketest.data.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface PokemonDAO {
    @Query("SELECT * FROM PokemonEntity")
    List<PokemonEntity> findAll();

    @Query("SELECT COUNT(*) FROM PokemonEntity WHERE id = :id")
    int countBy(String id);

    @Insert
    void save(PokemonEntity aPokemonEntity);
}

File: app/src/main/java/com/rmuhamed/sample/poketest/data/dao/PokemonDAO.kt
package com.rmuhamed.sample.poketest.data.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query

@Dao
interface PokemonDAO {
    @Query("SELECT * FROM PokemonEntity")
    suspend fun findAll(): List<PokemonEntity>

    @Query("SELECT COUNT(*) FROM PokemonEntity WHERE id = :id")
    suspend fun countBy(id: Int): Int

    @Insert
    suspend fun save(aPokemonEntity: PokemonEntity)

    @Query("SELECT * FROM PokemonEntity WHERE id = :id")
    suspend fun findBy(id: Int): PokemonEntity?
}